File: problems/03-sliding-window/002-minimum-size-subarray-sum/java/Solution.java
public class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0;
        int best = Integer.MAX_VALUE;
        long sum = 0;

        for (int right = 0; right < nums.length; right++) {
            sum += nums[right];

            while (sum >= target) {
                best = Math.min(best, right - left + 1);
                sum -= nums[left];
                left++;
            }
        }

        return best == Integer.MAX_VALUE ? 0 : best;
    }
}

File: problems/03-sliding-window/002-minimum-size-subarray-sum/python/solution.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        left = 0
        total = 0
        best = float("inf")

        for right, value in enumerate(nums):
            total += value

            while total >= target:
                best = min(best, right - left + 1)
                total -= nums[left]
                left += 1

        return 0 if best == float("inf") else best
